File: count_words/count.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strings"
)

type Count struct {
	Word string
	Count int
}

func main() {
	filename := os.Args[1]

	fmt.Println(filename)

	fh, err := os.Open(filename)
	if err != nil {
		fmt.Printf("Could not open file '%v': %v", filename, err)
		os.Exit(1)
	}

	reader := bufio.NewReader(fh)
	counter := make(map[string]int)

    for {
        line, _ := reader.ReadString('\n')
        //fmt.Print(line)
        fields := strings.Fields(line)
        //fmt.Println(fields)
        for _, word := range fields {
            word = strings.ToLower(word)
            counter[word]++
        }
        if line == "" {
            break
        }
    }

    words := make([]string, 0, len(counter))
    for word := range counter {
        words = append(words, word)
    }
    sort.Slice(words, func(i, j int) bool {
        return counter[words[i]] > counter[words[j]]
    })

    for _, word := range words {
        fmt.Printf("%v %v\n", word, counter[word])
    }
}

// ./count kjvbible.txt  0.07s user 0.02s system 106% cpu 0.080 total

File: count_words/count.rs
use std::io::{BufRead, BufReader};
use std::fs::File;
use std::collections::HashMap;


fn main() {
    // https://rust-cli.github.io/book/tutorial/cli-args.html
    let filename = std::env::args().nth(1).expect("no filename given.");

    let mut words: HashMap<String, u32> = HashMap::new();

    // https://stackoverflow.com/a/39434382
    let reader = BufReader::new(File::open(filename).expect("Cannot open file.txt"));
    for line in reader.lines() {
        for word in line.unwrap().split_whitespace() {
            // println!("word '{}'", word);
            // https://users.rust-lang.org/t/efficient-string-hashmaps-for-a-frequency-count/7752/2
            *words.entry(word.to_string()).or_insert(0) += 1;
        }
    }

    // println!("{:?}", words);
    // println!("{}", filename);

    let mut count_vec: Vec<_> = words.iter().collect();
    // https://stackoverflow.com/questions/34555837/sort-hashmap-data-by-value
    count_vec.sort_by(|a, b| a.1.cmp(b.1));
    // println!("{:?}", count_vec);
    count_vec.iter().for_each(|it| {
        println!("{}: {}", it.0, it.1)
    })

}

// ./count kjvbible.txt  0.45s user 0.02s system 99% cpu 0.467 total